src/lib/Component/Content/ContentEditMetaFieldsComponent.php

```diff
@@ -39,17 +40,30 @@ public function render(array $parameters = []): string
         $contentType = $parameters['content_type'];
 
         $metaFieldTypeIdentifiers = $this->getMetaFieldTypeIdentifiers();
+        $metaFieldGroups = $this->configResolver->getParameter(
+            'admin_ui_forms.content_edit.meta_fieldgroup_list'
+        );
+        $metaFieldDefinitionCollection = $contentType->fieldDefinitions->filter(
+            static fn (FieldDefinition $field): bool => true === in_array($field->fieldGroup, $metaFieldGroups),
+        );
 
-        if (empty($metaFieldTypeIdentifiers)) {
+        if (empty($metaFieldTypeIdentifiers) && $metaFieldDefinitionCollection->isEmpty()) {
             return self::NO_CONTENT;
         }
 
-        $parameters['meta_fields'] = [];
+        $metaFields = [];
+
+        foreach ($metaFieldDefinitionCollection as $fieldDefinition) {
+            $metaFields[] = $fieldDefinition->identifier;
+        }
```

Comment:
```suggestion
         $metaFields = array_column($metaFieldDefinitionCollection->toArray(), 'identifier');
```

This should be actually faster than manually iterating over elements in the loop. Given you need to merge results from both sources it's also a good candidate for new method.